packages/mint-components/src/components/sqm-coupon-code/sqm-coupon-code.feature

```diff
@@ -0,0 +1,129 @@
+
+
+  @motivating
+  Scenario: The first (i.e. most recently received) reward is the reward that is displayed
+    Given a user has at least one fueltank reward
+    And the fueltank reward is available
+    Then the coupon code component shows the first reward returned
+    And this reward is the most recently received reward by the user
+
+  @motivating
+  Scenario Outline: Coupon code has multiple states depending on reward status
+    Given a user has at least one fueltank reward
+    And the reward has most recent status <status>
+    And the dateScheduledFor field is <dateScheduledFor>
+    Then the component's status is set to <componentStatus>
+    And the <errorMessageTextProp> is displayed under the input field
+
+    Examples:
+      | status    | dateScheduledFor | componentStatus | errorMessageTextProp     |
+      | AVAILABLE | null             | AVAILABLE       | N/A                      |
+      | EXPIRED   | null             | EXPIRED         | expiredErrorMessage      |
+      | REDEEMED  | null             | REDEEMED        | redeemedMessage          |
+      | CANCELLED | null             | CANCELLED       | cancelledErrorMessage    |
+      | PENDING   | null             | EMPTY_TANK      | fullfillmentErrorMessage |
+      | PENDING   | 123412341234     | PENDING         | pendingErrorMessage      |
+      | null      | null             | ERROR           | genericErrorMessage      |
+
+  Scenario Outline: Coupon code's error message text props are grouped
+    Given an end user is viewing the coupon code component in the widget editor
+    And they are looking at the component's properties
+    Then they see <prop>
+    And <prop> is grouped under "Coupon code error"
+    Examples:
+      | prop                     |
+      | N/A                      |
+      | expiredErrorMessage      |
+      | redeemedMessage          |
+      | cancelledErrorMessage    |
+      | fullfillmentErrorMessage |
+      | pendingErrorMessage      |
+      | genericErrorMessage      |
+
+  @minutia
+  Scenario: Tooltip lifespan defaults to 2000
+    Given the tooltip's lifespan is set to 2000
+    And there is tooltip text
+    When the component renders
+    And the clipboard icon is clicked
+    Then a tooltip appears for ~2 seconds
+
+  @ui
+  Scenario: Component shows an error state when there is an error
+    Given a user is viewing the coupon code component
+    When there is an error in the coupon code
+    Then in place of the coupon code is an alert banner
+    And the alert banner gives information about the error to the user
+
+  @ui
+  Scenario: Component shows an loading state
+    Given a user is viewing the coupon code component
+    When the coupon code is loading in
+    Then the text inside the input is "Loading..."
+    And the coupon code input box has a gray background
+    And the cursor is set to "default"
+    And the user cannot copy the text
+
+  @ui
+  Scenario: user can edit the error message and code placeholder
```

Comment:
This really feels like an un-finished spec - what is the "then" here, what are we really trying to test?